packages/components/tests/lib/controllers/app-controller.test.js

```diff
@@ -1,839 +1,1389 @@
-import enUS from '@comp/locales/en-us/messages.json'
-import enGB from '@comp/locales/en-gb/messages.json'
-
-import { Constants } from 'alpheios-data-models'
-import { L10n } from 'alpheios-l10n'
-
-describe('ui-controller.test.js', () => {
-  console.error = function () {}
-  console.log = function () {}
-  console.warn = function () {}
-
-  let uiC, featureOptions, resourceOptions, state
-
-  beforeEach(async (done) => {
-    jest.spyOn(console, 'error')
-    jest.spyOn(console, 'log')
-    jest.spyOn(console, 'warn')
-    jest.spyOn(Options, 'initItems')
-
-    state = new State()
-    let sa1 = new LocalStorageArea('alpheios-feature-settings')
-    let sa2 = new LocalStorageArea('alpheios-resource-settings')
-    let sa3 = new LocalStorageArea('alpheios-ui-settings')
-    featureOptions = new Options(FeatureOptionDefaults, sa1)
-    resourceOptions = new Options(LanguageOptionDefaults, sa2)
-    let uiOptions = new Options(UIOptionDefaults, sa3)
-/*
-    uiC = new AppController(state, LocalStorageArea, {})
-    uiC.featureOptions = featureOptions
-    uiC.resourceOptions = resourceOptions
-    uiC.uiOptions = uiOptions
-
-    await uiC.init()
-    await uiC.activate()
-*/
-    done()
+import L10nModule from '@comp/vue/vuex-modules/data/l10n-module.js'
+import LexisModule from '@comp/vue/vuex-modules/data/lexis.js'
+import AuthModule from '@comp/vue/vuex-modules/data/auth-module.js'
+import Module from '@comp/vue/vuex-modules/module.js'
+import Locales from '@comp/locales/locales.js'
+import Platform from '@comp/lib/utility/platform.js'
+import Vue from '@vue-runtime'
+import MouseDblClick from '@comp/lib/custom-pointer-events/mouse-dbl-click.js'
+import LongTap from '@comp/lib/custom-pointer-events/long-tap.js'
+import GenericEvt from '@comp/lib/custom-pointer-events/generic-evt.js'
+import { Constants, LanguageModelFactory } from 'alpheios-data-models'
+import SelectionController from '@comp/lib/controllers/selection-controller.js'
+import { WordlistController } from 'alpheios-wordlist'
+import LexicalQuery from '@comp/lib/queries/lexical-query.js'
+import ResourceQuery from '@comp/lib/queries/resource-query.js'
+
+import BaseTestHelp from '@compTests/helpclasses/base-test-help.js'
+
+describe('AppController', () => {
+  let appC, uiState, state
+  const defaultMousemoveParams = {
+    enableMouseMoveLimitedByIdCheck: true,
+    mouseMoveAccuracy: 10,
+    mouseMoveDelay: 1000,
+    mouseMoveLimitedById: 'docs-editor-container'
+  }
+
+  /*
+  Since events are asynchronous, their listeners may be called in a test environment
+  after the application is tear down. To prevent errors related to this, a special
+  version of the create() method is created for the test environment. In it, event listeners
+  that may have issues like described above are not registered.
+   */
+  AppController.jestCreate = (state, options) => {
+    let appController = new AppController(state, options) // eslint-disable-line prefer-const
```

Comment:
Instead of having a test-specific create method couldn't we move the subscription calls to its own method in AppController with a parameter which indicates whether or not to skip the query subscriptions?